lib/galaxy/tools/parameters/basic.py

```diff
@@ -1702,7 +1702,7 @@ def single_to_json( value ):
     def to_python( self, value, app ):
         def single_to_python( value ):
             if isinstance( value, dict ) and 'src' in value:
-                id = app.security.decode_id( value[ 'id' ] )
+                id = value[ 'id' ] if isinstance( value[ 'id' ], int ) else app.security.decode_id( value[ 'id' ] )
```

Comment:
Are we willing to remove the decoding at some point in future? After all it was not present in code for long. Maybe a #TODO?